### PR TITLE
Library policies update

### PR DESCRIPTION
**Pull request by shelleydoljack**
@ndushay please review and merge when you get a chance. I created a new index using this branch on sw-solr-sirsi-test. Unfortunately, the SearchWorks app for this index (searchworks-test/morison) isn't up-to-date, so the library doesn't display in the record view, but you can see that the closed libraries are not indexed (example ckey 11403780).

### CODE REVIEW

File: stanford-sw/src/edu/stanford/Item.java [hunk 1]
```diff
@@ -94,7 +94,8 @@ public Item(DataField f999, String recId) {
 					|| StanfordIndexer.SKIPPED_LOCS.contains(homeLoc)
 					|| itemType.equals("EDI-REMOVE")
 					// SW-849 skip PHYSICS items that aren't location PHYSTEMP
-					|| (library.equals("PHYSICS") && (!homeLoc.equals("PHYSTEMP") && !currLoc.equals("PHYSTEMP"))))
+					|| (library.equals("PHYSICS") && (!homeLoc.equals("PHYSTEMP") && !currLoc.equals("PHYSTEMP")))
+          || StanfordIndexer.CLOSED_LIBS.contains(library))
```
Comment (ndushay) [inline stanford-sw/src/edu/stanford/Item.java]: the alignment looks weird ... but perhaps it's a tabs vs spaces from my Eclipse days?

File: stanford-sw/test/src/edu/stanford/ClosedLibrariesTests.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+		df999 = factory.newDataField("999", '0', '0');
+		df999.addSubfield(factory.newSubfield('a', "DO1 .A8133 V.402"));
+		df999.addSubfield(factory.newSubfield('w', "LC"));
+		df999.addSubfield(factory.newSubfield('c', "1"));
+		df999.addSubfield(factory.newSubfield('i', "36105210713654"));
+		df999.addSubfield(factory.newSubfield('k', "STACKS"));
+		df999.addSubfield(factory.newSubfield('m', "CHEMCHMENG"));
+    	df999.addSubfield(factory.newSubfield('t', "STKS"));
+    	rec.addVariableField(df999);
```
Comment (ndushay) [inline stanford-sw/test/src/edu/stanford/ClosedLibrariesTests.java]: more alignment fun here and below ...

File: stanford-sw/test/src/edu/stanford/ClosedLibrariesTests.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    // Test Case #3 Make sure record with multiple libraries including BIOLOGY and CHEMCHMENG in 999$m gets indexed
+      // but no building_facet for Biology or Chemistry/Chemical Eng
+    assertResultSize("id", "indexMultL", 1);
+
+		// Make sure no records have a building_facet belonging to a closed library
+		assertZeroResults("building_facet", "Biology (Falconer)");
+    assertZeroResults("building_facet", "Chemistry & ChemEng (Swain)");
+    assertZeroResults("building_facet", "Math & Statistics");
+	}
+}
```
Comment (ndushay) [inline stanford-sw/test/src/edu/stanford/ClosedLibrariesTests.java]: in a perfect world, you might consider separate test methods for the separate test cases:  #1, #2, #3, the facet ...

File: stanford-sw/test/src/edu/stanford/ItemInfoTests.java [hunk 4]
```diff
@@ -141,6 +138,11 @@ public void testSkipBuildingFacet()
 	    // INDEX-168 Meyer no longer exists
 	    assertZeroResults(fldName, "\"MEYER\"");
 
+      // BIOLOGY, CHEMCHMENG, and MATH-CS are closed Libraries
+      assertZeroResults(fldName, "\"BIOLOGY\"");
+      assertZeroResults(fldName, "\"CHEMCHMENG\"");
+      assertZeroResults(fldName, "\"MATH-CS\"");
```
Comment (ndushay) [inline stanford-sw/test/src/edu/stanford/ItemInfoTests.java]: alignment, and you can keep or remove the blank line before the close brace.  Not sure why it was in there before ...

File: stanford-sw/translation_maps/library_map.properties [hunk 5]
```diff
@@ -3,9 +3,9 @@
 #APPLIEDPHY = Applied Physics Department
 ARS = Archive of Recorded Sound
 ART = Art & Architecture (Bowes)
-BIOLOGY = Biology (Falconer)
+# BIOLOGY = Biology (Falconer) closed library 12/2016
```
Comment (ndushay) [inline stanford-sw/translation_maps/library_map.properties]: 👍 like how you did this commenting out

File: stanford-sw/test/data/closedLibraries.xml [hunk 6]
```diff
@@ -0,0 +1,122 @@
+<records xmlns='http://www.loc.gov/MARC21/slim'>
```
Comment (ndushay) [inline stanford-sw/test/data/closedLibraries.xml]: in a perfect world, I prefer record creation inline in the test so you can see what data you are testing in one shot.  Not worth redoing here.  Just something to keep in mind for future new test data.  

If there had been time, I would have gotten rid ofall the marc data test files and put it all inline.  :-P

**Review by ndushay — COMMENTED** [review]
In general, this is fine as is.   

- Alignment does help with code readability ... 
Not sure what editor you are using, but if you convert any files you are editing to use spaces instead of tabs and get good alignment, I think it would be a win.   I was NOT good about this myself as i worked on this alone in Eclipse and took spaces or tabs as they came.  But I think that was a mistake.

- I would also suggest grouping conceptual work into a single commit.  
Making a change and writing the tests for the change is all part of the same concept.  Not a deal breaker, but a good habit to form.  If you don't know how to do interactive rebasing in git, I would be happy to show you.  I use it all the time to try to clean up my actual commits into commits that work better for a PR.
  so in a perfect world, I would combine 5da8819, 63d0357 and 8fed54d into a single commit, for example.

**Comment by ndushay** [comment]
So basically, 👍 for you to merge if you don't want to muck about with all the small stuff I've pointed out.

File: stanford-sw/src/edu/stanford/Item.java (hunk 1 above)
Comment (shelleydoljack) [inline stanford-sw/src/edu/stanford/Item.java]: Oh, yes. I've got my Atom configured to insert 2 spaces when I hit tab but it seems all the previous code uses tabs. Is there an easy way to convert the tabs to spaces?

**Comment by shelleydoljack** [comment]
I see how to convert tabs to spaces in Atom. I'll work on doing that for the next round of work.

File: stanford-sw/src/edu/stanford/Item.java (hunk 1 above)
Comment (ndushay) [inline stanford-sw/src/edu/stanford/Item.java]: In Atom, In the Packages menu, I have a "whitespace" thing I can use to convert tabs to whitsepace.  It seems to miss a few tabs sometimes.  I'm not sure if that came with Atom or if I got an add on.

**Pull request merged**